Cypress commands refactor

MarkvdVorst opened the pull request:
Closes #511 

Converting all tests to typescript files.
Added intellisence to the tests for the custom cypress commands.
Added function for interceptGetApiCall to adhere to DRY rule.
Put cypress on ignore for eslint.
Refactored tests where I noticed some inconsistencies.

MatthijsSmets commented on cypress/support/commands.ts:
```diff
@@ -222,34 +187,45 @@ Cypress.Commands.add('clickFirstFileInFileTree' as keyof Chainable, () => {
     : cy.getTableBody().find('tr').should('have.length', length);
 });
+
+function interceptGetApiCall(url: StringMatcher, alias: string): void {
```
I see this method is called a lot of times with: `/\/api\/*?/g` as url, maybe it's better to have that as default value for url?

MarkvdVorst commented:
Added types where possible, created an interface for API response to add a type where an API response was expected and changed a foreach to a more conventional for of loop.
